### PR TITLE
Re-initialize the BIOS DAC callbacks on SB and Tandy changes

### PR DESCRIPTION
**Pull request by kcgen**
# Description

When changing the Sound Blaster or Tandy DAC settings, re-initialize the BIOS DAC callbacks to match the new configuration.

## Related issues

Reported by **Tazman** on the eXoDOS Discord channel:

> _"When I was in my teens, I had a Tandy 1000 TL/2 computer and it supported 3 voice sound. I'm trying to get SQ3 in dosbox to duplicate the sound effects using tandy sound. It plays the music right, but not the sound effects like when Roger comes out of sleeping chamber. You hear the glass shroud slide back and Roger actually says "Where am I?".  I know I can just use any of the other sound drivers like soundblaster and such in exodos. Which is really awesome to have so much flexibility playing SQ3, but for my nostalgia fix, I would like to reproduce what I once played."_

# Manual testing

Confirmed that `tandy on` can be set on-the-fly, and the DAC effects (described above) in Space Quest 3 play as expected (skip to 3:20):

https://github.com/dosbox-staging/dosbox-staging/assets/1557255/a3da8ae4-04c5-4413-a809-13b33749e837

Testing pack: [sq3.zip](https://github.com/dosbox-staging/dosbox-staging/files/13406530/sq3.zip)

Regression tested a handful of Sound Blaster and Tandy games using on-the-fly changes and confirmed they all still play as expected.

Another excuse to post Ron Hubbard's ripping tunes :metal:

https://github.com/dosbox-staging/dosbox-staging/assets/1557255/231e493d-0371-44b4-b546-871487ce653e

Sanitizer build is also passing without issues.

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [x] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by johnnovak** [comment]
Super cool enhancement, @kcgen 🎉 For the record, I'm pretty sure the Rob Hubbard tune in the second video only uses the Tandy PSG for sample playback. If I remember correctly, that confused me when finding games that exercise the Tandy DAC. Rob just used some tricks to play back sampled sounds with the PSG alone! (he used to do similar tricks on the C64 that doesn't even have a DAC)

File: src/ints/bios.cpp [hunk 1]
```diff
@@ -1032,17 +1032,117 @@ void BIOS_ZeroExtendedSize(bool in) {
...
+		tandy_DAC_callback[1] = nullptr;
+	}
+	tandy_sb.port  = 0;
+	tandy_dac.port = 0;
+}
+
+void BIOS_SetupTandySbDacCallbacks()
+{
+	/* tandy DAC can be requested in tandy_sound.cpp by initializing this
+	 * field */
```
Comment (johnnovak) [inline src/ints/bios.cpp:1075]: Minor: maybe just use `//` comments everywhere?

File: src/ints/bios.cpp [hunk 2]
```diff
@@ -1032,17 +1032,117 @@ void BIOS_ZeroExtendedSize(bool in) {
...
+	const bool use_tandyDAC = (real_readb(0x40, 0xd4) == 0xff);
```
Comment (johnnovak) [inline src/ints/bios.cpp:1076]: `use_tandy_dac`

**Review by johnnovak — APPROVED** [review]
Nice one @kcgen , just some extremely minor comments 🎉

File: src/ints/bios.cpp (hunk 1 above)
Comment (kcgen) [reply src/ints/bios.cpp:1075]: Switched them over (the code here is existing and just relocated verbatim)

**Comment by kcgen** [comment]
> For the record, I'm pretty sure the Rob Hubbard tune in the second video only uses the Tandy PSG for sample playback.

Yup! The guitar specifically really pushes the limits of the PSG :-) 
This was was tested w/ `tandy = psg`; just part of random regression testing.

**Pull request merged**